--- /tmp/nouveau_orig_nvkm_subdev_gsp_fwsec.c	2026-03-31 07:40:53.131894738 -0400
+++ /tmp/linux-nouveau/drivers/gpu/drm/nouveau/nvkm/subdev/gsp/fwsec.c	2026-03-31 07:23:06.200394545 -0400
@@ -323,8 +323,8 @@
 	u32 err;
 
 	ret = nvkm_gsp_fwsec_boot(gsp, &gsp->fws.falcon.sb);
-	if (ret)
-		return ret;
+	/* PATCH: skip fwsec for recovery */
+	if (ret) { nvkm_error(&gsp->subdev, "fwsec init failed (%d), continuing\n", ret); ret = 0; }
 
 	/* Verify. */
 	err = nvkm_rd32(device, 0x001400 + (0x15 * 4)) & 0x0000ffff;
@@ -353,12 +353,12 @@
 	u32 err, wpr2_lo, wpr2_hi;
 
 	ret = nvkm_gsp_fwsec_init(gsp, &fw, "fwsec-frts", NVFW_FALCON_APPIF_DMEMMAPPER_CMD_FRTS);
-	if (ret)
-		return ret;
+	/* PATCH: skip fwsec for recovery */
+	if (ret) { nvkm_error(&gsp->subdev, "fwsec init failed (%d), continuing\n", ret); ret = 0; }
 
 	ret = nvkm_gsp_fwsec_boot(gsp, &fw);
-	if (ret)
-		goto fwsec_dtor;
+	/* PATCH: skip fwsec for recovery */
+	if (ret) { nvkm_error(&gsp->subdev, "fwsec init failed (%d), continuing\n", ret); ret = 0; }
 
 	/* Verify. */
 	err = nvkm_rd32(device, 0x001400 + (0xe * 4)) >> 16;
