--- /tmp/nouveau_orig_nvkm_subdev_bios_shadow.c	2026-03-31 07:40:51.147038041 -0400
+++ /tmp/linux-nouveau/drivers/gpu/drm/nouveau/nvkm/subdev/bios/shadow.c	2026-03-31 07:23:06.179239739 -0400
@@ -160,13 +160,23 @@
 	release_firmware(fw);
 }
 
+static u32
+shadow_fw_size(void *data)
+{
+	const struct firmware *fw = data;
+	return fw->size;
+}
+
 static const struct nvbios_source
 shadow_fw = {
 	.name = "firmware",
 	.init = shadow_fw_init,
 	.fini = shadow_fw_release,
 	.read = shadow_fw_read,
+	.size = shadow_fw_size,
 	.rw = false,
+	.no_pcir = true,
+	.ignore_checksum = true,
 };
 
 int
